packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -268,7 +268,7 @@ export function useGridEditRows(apiRef: GridApiRef) {
 
   const handleCellKeyDown = React.useCallback(
     (params: GridCellParams, event) => {
-      if (event.isPropagationStopped()) {
+      if (!params.isEditable || event.isPropagationStopped()) {
```

Comment:
This makes me think about https://github.com/mui-org/material-ui-x/issues/1300#issuecomment-808742299. We could likely do a follow up with them.